feat: make Select receive custom options

RyanZhiNie opened the pull request:
## 📑 Description
In the current version, `items` prop's value can only be an array of `{name, value}` and then the array is transformed into options. But these options cannot be customized, e.g., `selected` or `disabled`.
I make `"slot"` be the alternative value, so users can pass their desired option elements into this component.

## ✅ Checks
- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [ ] I have checked the page with https://validator.unl.edu/
- [ ] All the tests have passed
- [x] My pull request is based on the latest commit (not the npm version).

jjagielka commented:
That request make sense. Couple of suggestions though:
1. `export let items: SelectOptionType[] | 'slot' = [];`
This is completely unneeded. Better approach is to use `else` in the `each` loop that will trigger when `items` are empty.
```svelte
  {#each items as { value, name }}
    <option {value}>{name}</option>
  {:else}
    <slot />
  {/each}
```
2. `placeholder` should be independent from `items`. If we want to make it optional then:
```svelte
  {#if placeholder}
    <option disabled selected value="">{placeholder}</option>
  {/if}
```
This will force to change your example in docs to:
```svelte
<Select id="countries" class="mt-2" bind:value={selected} placeholder="">
```
3. Why did you change the lines 60-61? It's not connected with what you want to achieve.

RyanZhiNie commented:
Nice suggestions. I've updated my changes. Thx.

> Why did you change the lines 60-61? It's not connected with what you want to achieve.

I changed these two lines because the html codes they generated were inconsistent with the code shown in flowbite website, but you're right, they are not connected with my PR, maybe I will make another PR to fix them.

jjagielka commented:
@shinokada - I'm OK with that PR
